WalletWasabi/Tor/Socks5/Models/Messages/MethodSelectionResponse.cs

```diff
@@ -6,21 +7,22 @@ namespace WalletWasabi.Tor.Socks5.Models.Messages
 {
 	public class MethodSelectionResponse : ByteArraySerializableBase
 	{
-		public MethodSelectionResponse()
+		/// <param name="bytes">2 bytes are required to be passed in.</param>
+		public MethodSelectionResponse(byte[] bytes)
 		{
+			Guard.Same($"{nameof(bytes)}.{nameof(bytes.Length)}", 2, bytes.Length);
+
+			Ver = new VerField(bytes[0]);
+			Method = new MethodField(bytes[1]);
 		}
 
-		public VerField Ver { get; set; }
+		public VerField Ver { get; }
 
-		public MethodField Method { get; set; }
+		public MethodField Method { get; }
 
 		public override void FromBytes(byte[] bytes)
 		{
-			Guard.NotNullOrEmpty(nameof(bytes), bytes);
-			Guard.Same($"{nameof(bytes)}.{nameof(bytes.Length)}", 2, bytes.Length);
-
-			Ver = new VerField(bytes[0]);
-			Method = new MethodField(bytes[1]);
+			throw new NotSupportedException("This method is no longer supported.");
```

Comment:
NACK

Reply:
It would be more productive to say: "This is bad because ..." as it would not make me guess what you don't like and ultimately waste your time less over longer timeframe.

Reply:
This is bad, because there are turtles all the way down. If you only mess with a single thing here, then it's just awkward.

Reply:
True. Ok, next time, I'll change it completely.